bot-components/GitHub_GraphQL.ml

```diff
@@ -124,6 +124,34 @@ module Issue_Milestone =
   }
 |}]
 
+module MergePullRequestInfo =
+[%graphql
+{|
+  query mergePullRequestInfo($owner: String!, $repo: String!, $number: Int!) {
+    repository(owner: $owner, name: $repo) {
+      pullRequest(number: $number) {
+        baseRef {
+          name
+        }
+        files(first: 100) {
+          nodes {
+            path
+          }
+        }
+        reviewDecision
+        reviews(last: 100) {
```

Comment:
I suggest requesting separately the approved reviews and commented reviews in order to make sure we don't miss approved reviews, as in:

```
      commentReviews: reviews(states: COMMENTED, last: 100) { ... }
      approvedReviews: reviews(states: APPROVED, last: 100) { ... }
```